cmd/query/app/ui/assets.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package ui
+
+import (
+	"embed"
+	"net/http"
+
+	"github.com/jaegertracing/jaeger/pkg/gzipfs"
+	"github.com/jaegertracing/jaeger/pkg/httpfs"
+)
+
+//go:embed actual/*
+var actualAssetsFS embed.FS
+
+//go:embed placeholder/index.html
+var placeholderAssetsFS embed.FS
+
+// GetStaticFiles gets the static assets that the Jaeger UI will serve. If the actual
+// assets are available, then this function will return them. Otherwise, a
+// non-functional index.html is returned to be used as a placeholder.
+func GetStaticFiles() http.FileSystem {
+	if _, err := actualAssetsFS.ReadFile("actual/index.html.gz"); err != nil {
+		return httpfs.PrefixedFS("placeholder", http.FS(placeholderAssetsFS))
+	}
+
+	return httpfs.PrefixedFS("actual", http.FS(gzipfs.New(actualAssetsFS)))
+}
```

Comment:
is there a way to write a unit test for this?

Reply:
Sure, why not. The test should be non-destructive, so if the file already exists, move it to temp dir and then move back as test cleanup function.

Reply:
@yurishkuro I gave this a shot and couldn't get it to work since I think the embed directives get executed at build time so any file system changes I make in the test body don't get picked up. Let me know if there's something I'm doing wrong.